### PR TITLE
Add documentation for user data downloads

### PR DESCRIPTION
**Pull request by Ninjaclasher**
Depends on DMOJ/online-judge#1351.

### CODE REVIEW

File: docs/site/user_data_download.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# Path to use for nginx's X-Accel-Redirect feature.
+# Should be an internal location mapped to the above directory.
+DMOJ_USER_DATA_INTERNAL = '/datacache'
+# How often a user can download their data.
+DMOJ_USER_DATA_DOWNLOAD_RATELIMIT = datetime.timedelta(days=1)
+```
+
+Also, uncomment the revelant section in your Nginx configuation if you wish to take 
+advantage of Nginx's X-Accel-Redirect feature.
+```
```
Comment (TheAvidDev) [inline docs/site/user_data_download.md:24]: \`\`\`nginx

Comment (Ninjaclasher) [reply docs/site/user_data_download.md]: Done

File: docs/site/user_data_download.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+DMOJ_USER_DATA_CACHE = '/home/dmoj-uwsgi/datacache'
+
+# Path to use for nginx's X-Accel-Redirect feature.
+# Should be an internal location mapped to the above directory.
+DMOJ_USER_DATA_INTERNAL = '/datacache'
+# How often a user can download their data.
+DMOJ_USER_DATA_DOWNLOAD_RATELIMIT = datetime.timedelta(days=1)
+```
+
+Also, uncomment the revelant section in your Nginx configuation if you wish to take
```
Comment (Xyene) [inline docs/site/user_data_download.md:22]:
```suggestion
Also, uncomment the relevant section in your Nginx configuration if you wish to take 
```

File: docs/site/user_data_download.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+# Path to use for nginx's X-Accel-Redirect feature.
+# Should be an internal location mapped to the above directory.
+DMOJ_USER_DATA_INTERNAL = '/datacache'
+# How often a user can download their data.
+DMOJ_USER_DATA_DOWNLOAD_RATELIMIT = datetime.timedelta(days=1)
+```
+
+Also, uncomment the revelant section in your Nginx configuation if you wish to take 
+advantage of Nginx's X-Accel-Redirect feature.
```
Comment (Xyene) [inline docs/site/user_data_download.md:23]:
```suggestion
advantage of Nginx's `X-Accel-Redirect` feature.
```

maybe link to docs?

File: docs/site/user_data_download.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+These data files are not cleaned up automatically. Although each user can have at most one data file 
+stored on the server at a time, you may want to clean up old files. A Cron job should suffice:
+```
+0 */4 * * * find /home/dmoj-uwsgi/datacache/ -type f -mtime 2 -delete
+```
+
+This Cron job will delete files older than 2 days every 4 hours. You are recommended to tweak these 
+values according to your ratelimit.
+
+You should now find a link on your Edit Profile that allows you to download your data,
```
Comment (Xyene) [inline docs/site/user_data_download.md:45]:
```suggestion
You should now find a link on your _Edit Profile_ that allows you to download your data, 
```

**Comment by Ninjaclasher** [comment]
This has been updated.

**Pull request merged**